legion/engine/physics/mesh_splitter_utils/intersecting_polygon_organizer.hpp

```diff
@@ -0,0 +1,660 @@
+#pragma once
+#include <core/core.hpp>
+#include <physics/mesh_splitter_utils/splittable_polygon.h>
+#include <physics/mesh_splitter_utils/mesh_splitter_typedefs.hpp>
+#include <physics/mesh_splitter_utils/mesh_splitter_debug_helpers.h>
+#include <physics/mesh_splitter_utils/intersection_edge_info.h>
+
+namespace legion::physics
+{
+    /** @struct IntersectingPolygonOrganizer
+    * @brief Contains the necessary operations to split the edges of a given polygon
+    */
+    struct IntersectingPolygonOrganizer
+    {
+        MeshSplitterDebugHelper* debugHelper = nullptr;
+
+        IntersectingPolygonOrganizer(MeshSplitterDebugHelper* pDebugHelper) : debugHelper(pDebugHelper)
+        {
+
+        }
+
+        //------------------------------------------------------ Related to Edge Categorization  ----------------------------------------------------------------//
```

Comment:
Instead of comments like this you could do
```cpp
#prama region regionName
//All of the beautiful code
#pramga endregion
```
this would also allow you to collapse the region